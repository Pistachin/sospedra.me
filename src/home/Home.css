.menu {
  left: 50%;
  min-width: 320px; /* iPhone 5s screen */
  position: absolute;
  top: 38%; /* golden ratio */
  text-align: center;
  transform: translate(-50%, -38%);
  width: 50%;
  z-index: 10;
}

.move-slide {
  animation-duration: 5s; /* default value */
  animation-fill-mode: forwards;
  animation-timing-function: ease-in-out;
  animation-name: slide-left;
}

.sprite {
  background-size: contain;
  bottom: 0;
  height: 50%;
  left: 0;
  position: absolute;
  width: 250%;
  margin: 0;
}

.title {
  font-family: lazer84;
  color: #6df7ea;
  font-size: 42px;
  text-shadow: 1.9px 2px 0 #3f2d43;
}

.title a {
  color: inherit;
  text-decoration: none;
}

.sub-title {
  color: #ea1195;
  text-shadow: -1px 1px 0 #ffb9b9;
}

.kanji {
  font-size: 32px;
  text-shadow: -1px 1px 0 #184854;
}

@keyframes slide-left {
  0% { transform: translateX(0); }
  100% { transform: translateX(-50%); }
}
