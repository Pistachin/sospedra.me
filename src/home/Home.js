import Inferno from 'inferno'
import { Link } from 'inferno-router'
import './Home.css'

const BG_SPRITE = 'images/street-sprite.svg'

const Home = () => (
  <div>
    <main className='menu'>
      <h1 className='title'>
        Rub&eacute;n Sospedra
      </h1>
      <h2 className='sub-title'>
        javascript hacker * front-end developer
      </h2>
    </main>

    <figure
      className='sprite move-slide'
      alt='retrowave-skyline-moving-left-as-background'
      style={{
        animationDuration: `${window.innerWidth / 138}s`,
        backgroundImage: `url(${process.env.PUBLIC_URL}/${BG_SPRITE})`
      }}
    />
  </div>
)

export default Home
