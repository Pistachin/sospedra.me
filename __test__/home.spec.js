/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Home from './pages/index.js'

describe('Home', () => {
  it('shows titles and sprite', () => {
    const wrapper = renderer.create(<Home />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
