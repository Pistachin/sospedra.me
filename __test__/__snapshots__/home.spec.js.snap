// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home shows titles and sprite 1`] = `
<div
  className="jsx-1346193646 root"
>
  <main
    className="jsx-1346193646"
  >
    <h1
      className="jsx-1660294369"
    >
      <a
        className="jsx-1660294369"
        href="https://twitter.com/sospedra_r"
        target="_blank"
      >
        Rubén Sospedra
      </a>
    </h1>
    <h2
      className="jsx-2712971757"
    >
      ｊａｖａｓｃｒｉｐｔ    ｈａｃｋｅｒ
      <br
        className="jsx-2712971757"
      />
      ｆｕｌｌｓｔａｃｋ ｅｎｇｉｎｅｅｒ
    </h2>
  </main>
  <figure
    alt="retrowave-skyline-moving-left-as-background"
    className="jsx-77113858"
  />
</div>
`;
